### ISSUE TITLE
GGM - BUG - If lights aren't set to "cast shadows", they bleed through solid walls

### ISSUE THREAD

**Issue by Kravenwolf** [TheGameCreators/GameGuruRepo#4285]
https://user-images.githubusercontent.com/103399301/227104742-c67ba250-8d2b-48e8-9730-d62aee158280.mp4

**Comment by MonkeyFrogStudio** [comment]
I don't believe this is actually a bug. If Cast Shadow is not ticked, then that tells the engine that objects don't interact with it (casting a shadow). Walls, as far as the engine is concerned, are just 'objects', like any other object in the game. If you had a barrel in the scene and a light near it with Cast Shadow unticked, then the light would "leak" through the barrel to the other side. A wall is no different.

What you need to do in this case, if you require a non-shadow casting light to be "blocked", is to use a light with a smaller radius. And if it needs to cover a larger area, then you will need to use more lights. Else, you will need to use a shadow casting light.

**Comment by davetgc** [comment]
As Monkeyfrog has said, I'm not sure this is a bug, but expected behavior.

**Comment by davetgc** [comment]
Closed, but it's been useful as a how to guide :)

**Comment by MonkeyFrogStudio** [comment]
@Kravenwolf Glad to have helped in some small way. :)

Part of level design planning is also planning for these areas where you'll use large lights that don't cast shadows. Make sure there are large enough gaps between walls (walls that separate rooms) so that light doesn't "spill" over from the lit area, etc. 

BTW - I don't believe that non-shadow casting lights will be too intensive on your system. You should be able to get away with quite a few of them ... a heck of a lot more than shadow casting lights. Even when I had an older card, I had dozens and dozens of non-shadow casting lights in my levels and wasn't noticing a frame rate hit.
